Packs/FortiGate/Integrations/FortiGate/FortiGate.py

```diff
@@ -1,93 +1,1493 @@
+            vdom (str, optional): The VDOM to use.
+                Defaults to VDOM_DEFAULT.
+
+        Returns:
+            dict[str, Any]: The API response.
+        """
+        return self._http_request(
+            method="DELETE",
+            url_suffix=urljoin(self.POLICY_ENDPOINT, id_),
+            params={"vdom": vdom},
+            error_handler=Client._error_handler,
+        )
+
+    def list_system_vdoms(
+        self,
+        filter_field: str | None = None,
+        filter_value: str | None = None,
+        format_fields: list[str] | None = None,
+    ) -> dict[str, Any]:
+        """List all virtual domains.
+
+        Args:
+            filter_field (str | None, optional): The field to filter by.
+                Defaults to None.
+            filter_value (str | None, optional): The value to filter by.
+                Defaults to None.
+            format_fields (list[str], optional): The fields to format.
+                Defaults to None.
+
+        Returns:
+            dict[str, Any]: The API response.
+        """
+        return self._http_request(
+            method="GET",
+            url_suffix="cmdb/system/vdom",
+            params=remove_empty_elements(
+                {
+                    "filter": self._get_filter(filter_field, filter_value),
+                    "format": self._get_format(format_fields),
+                }
+            ),
+            error_handler=Client._error_handler,
+        )
+
+    def list_banned_ips(
+        self,
+        vdom: str | None = DEFAULT_VDOM,
+        filter_field: str | None = None,
+        filter_value: str | None = None,
+        format_fields: list[str] | None = None,
+    ) -> dict[str, Any]:
+        """List all banned IPv4 and IPv6 addresses.
+
+        Args:
+            vdom (str, optional): The VDOM to use.
+                Defaults to VDOM_DEFAULT.
+            filter_field (str | None, optional): The field to filter by.
+                Defaults to None.
+            filter_value (str | None, optional): The value to filter by.
+                Defaults to None.
+            format_fields (list[str], optional): The fields to format.
+                Defaults to None.
+
+        Returns:
+            dict[str, Any]: The API response.
+        """
+        return self._http_request(
+            method="GET",
+            url_suffix=urljoin(self.BANNED_IP_ENDPOINT, "select"),
+            params=remove_empty_elements(
+                {
+                    "vdom": vdom,
+                    "filter": self._get_filter(filter_field, filter_value),
+                    "format": self._get_format(format_fields),
+                }
+            ),
+            error_handler=Client._error_handler,
+        )
+
+    def ban_ip(self) -> dict[str, Any]:
+        pass
+
+    def unban_ip(self) -> dict[str, Any]:
+        pass
+
+
+""" Helper Commands  """
+
+
+def map_keys(old_dict: dict[str, Any], mappings: list[Mapping]) -> dict[str, Any]:
```

Comment:
This function has many test cases through `LIST` commands, because each response and output file are used especially to test this logic, is it really needed?